src/App.tsx

```diff
@@ -1,34 +1,105 @@
-/* eslint-disable max-len */
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
+import { getTodos, getUser } from './api';
+
+import { User } from './types/User';
+import { Todo } from './types/Todo';
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [isLoading, setIsLoading] = useState(false);
+  const [selectedTodo, setSelectedTodo] = useState<Todo | null>(null);
+  const [user, setUser] = useState<User | null>(null);
+  const [isUserLoading, setIsUserLoading] = useState<boolean>(false);
+  const [filter, setFilter] = useState<string>('all');
+  const [query, setQuery] = useState<string>('');
+
+  const handleToDoClick = (userId: number) => {
+    setIsUserLoading(true);
+    getUser(userId).then((userData: User) => {
+      setUser(userData);
+      setIsUserLoading(false);
+    }).catch(() => {
+      setIsUserLoading(false);
+    });
+  };
+
+  const handleClearQuery = () => {
+    setQuery('');
+  };
+
+  useEffect(() => {
+    setIsLoading(true);
+    getTodos().then((data: Todo[]) => {
+      setTodos(data);
+      setIsLoading(false);
+    }).catch(() => {
+      setIsLoading(false);
+    });
+  }, []);
+
+  const handleShowClick = (todo: Todo) => {
+    setSelectedTodo(todo);
+    handleToDoClick(todo.userId);
+  };
+
+  const filteredTodos = todos.filter((todo) => {
+    if (query && !todo.title.toLowerCase().includes(query.toLowerCase())) {
+      return false;
+    }
+
+    switch (filter) {
+      case 'completed':
+        return todo.completed;
+      case 'active':
+        return !todo.completed;
+      default:
+        return true;
+    }
+  });
```

Comment:
please move it outside of your component, it doesn't need to be there, and it makes your component bigger and bigger